checkov/common/runners/runner_registry.py

```diff
@@ -101,8 +102,11 @@ def _parallel_run(runner: _BaseRunner) -> Report | list[Report]:
             reports = parallel_runner.run_function(func=_parallel_run, items=self.runners, group_size=1)
 
         merged_reports = self._merge_reports(reports)
-        if bc_integration.bc_api_key:
-            SecretsOmitter(merged_reports).omit()
+
+        if self.secrets_omitter_class_name and bc_integration.bc_api_key:
+            secrets_omitter_class_obj = globals().get(self.secrets_omitter_class_name)
+            if secrets_omitter_class_obj:
+                secrets_omitter_class_obj(merged_reports).omit()
```

Comment:
@Eliran-Turgeman if you pass in the class, then:
```suggestion
        if self.secrets_omitter_class_name and bc_integration.bc_api_key:
            self.secrets_omitter_class_name(merged_reports).omit()
```